src/impls/create.rs

```diff
@@ -2,70 +2,61 @@ use crate::Counter;
 
 use num_traits::{One, Zero};
 
-use std::hash::Hash;
-use std::iter;
 use std::ops::AddAssign;
+use std::collections::HashMap;
+use std::hash::Hash;
 
-impl<T, N> Default for Counter<T, N>
+impl<T, N> Counter<T, N>
 where
     T: Hash + Eq,
-    N: Default,
+    N: Zero,
 {
-    fn default() -> Self {
-        Self {
-            map: Default::default(),
-            zero: Default::default(),
+    /// Create a new, empty `Counter`
+    pub fn new() -> Counter<T, N> {
+        Counter {
+            map: HashMap::new(),
+            zero: N::zero(),
         }
     }
 }
 
-impl<T, N> iter::FromIterator<T> for Counter<T, N>
+impl<T, N> Counter<T, N>
 where
     T: Hash + Eq,
     N: AddAssign + Zero + One,
 {
-    /// Produce a `Counter` from an iterator of items. This is called automatically
-    /// by [`Iterator::collect()`].
-    ///
-    /// [`Iterator::collect()`]:
-    /// https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.collect
-    ///
-    /// ```rust
-    /// # use counter::Counter;
-    /// # use std::collections::HashMap;
-    /// let counter = "abbccc".chars().collect::<Counter<_>>();
-    /// let expect = [('a', 1), ('b', 2), ('c', 3)].iter().cloned().collect::<HashMap<_, _>>();
-    /// assert_eq!(counter.into_map(), expect);
-    /// ```
-    ///
-    fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
-        Counter::<T, N>::init(iter)
+    /// Create a new `Counter` initialized with the given iterable.
+    pub fn init<I>(iterable: I) -> Counter<T, N>
```

Comment:
`init` and `update` semantically create a counter from an iterator, so should live in `from_iterator.rs`.

Reply:
yes i just realized that about update as well